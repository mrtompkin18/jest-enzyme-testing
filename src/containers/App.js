import React from 'react'
import { Card } from 'antd'
import Todo from './Todo'
import TodoList from './TodoList'

class App extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            todos: [{
                taskId: 1,
                task: 'Go to Home',
                isEdit: false
            }],
            input: undefined,
            id: 1
        }
    }

    onInputChange = (e) => {
        this.setState({ input: e.target.value })
    }

    addTask = () => {
        const taskId = this.state.id + 1
        const task = this.state.input
        this.setState({
            todos:
                [...this.state.todos, {
                    taskId,
                    task
                }],
            input: undefined,
            id: taskId
        })
    }

    deleteTask = (taskId) => {
        const updatedTodos = this.state.todos.filter(item => item.taskId !== taskId)
        this.setState({ todos: updatedTodos })
    }

    editTask = (item) => {
        item.isEdit = true
        this.setState({ todos: this.state.todos })
    }

    cancelEditTask = (item) => {
        item.isEdit = false
        this.setState({ todos: this.state.todos })
    }

    saveTask = (item, newText) => {
        if (newText !== undefined && newText !== '') {
            item.task = newText
        }
        item.isEdit = false
        this.setState({ todos: this.state.todos })

    }

    render() {
        return (
            <Card title="Jest Enzyme">
                <Todo
                    input={this.state.input}
                    addTask={this.addTask}
                    onInputChange={this.onInputChange}
                />
                <TodoList
                    todos={this.state.todos}
                    editTask={this.editTask}
                    deleteTask={this.deleteTask}
                    cancelEditTask={this.cancelEditTask}
                    saveTask={this.saveTask}
                />
            </Card>
        )
    }
}
export default App