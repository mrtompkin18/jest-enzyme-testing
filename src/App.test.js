import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import { configure } from 'enzyme'
import { shallow } from 'enzyme'
import App from './containers/App'

configure({ adapter: new Adapter() });
describe('Todos Testing', () => {
  it('start taskId with 1 ', () => {
    const wrap = shallow(<App />)
    expect(wrap.state().id).toEqual(1)
  })

  it('matches snapshot', () => {
    const tree = renderer.create(<App />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})