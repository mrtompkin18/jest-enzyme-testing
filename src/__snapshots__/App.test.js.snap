// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos Testing matches snapshot 1`] = `
<div
  className="ant-card ant-card-bordered"
>
  <div
    className="ant-card-head"
  >
    <div
      className="ant-card-head-wrapper"
    >
      <div
        className="ant-card-head-title"
      >
        Jest Enzyme
      </div>
    </div>
  </div>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-row"
      style={
        Object {
          "marginLeft": -4,
          "marginRight": -4,
        }
      }
    >
      <div
        className="ant-col-22"
        style={
          Object {
            "paddingLeft": 4,
            "paddingRight": 4,
          }
        }
      >
        <input
          className="ant-input ant-input-lg"
          disabled={false}
          id="task"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Please input your task"
          size="large"
          type="text"
          value=""
        />
      </div>
      <div
        className="ant-col-2"
        style={
          Object {
            "paddingLeft": 4,
            "paddingRight": 4,
          }
        }
      >
        <button
          className="ant-btn ant-btn-lg"
          id="addButton"
          onClick={[Function]}
          style={
            Object {
              "float": "right",
              "width": "100%",
            }
          }
          type="button"
        >
          <span>
            Add
          </span>
        </button>
      </div>
    </div>
    <div>
      <br />
      <div
        className="ant-list ant-list-split"
      >
        <div
          className="ant-spin-nested-loading"
          style={null}
        >
          <div
            className="ant-spin-container"
          >
            <div
              className="ant-list-item"
            >
              <div
                className="ant-list-item-content ant-list-item-content-single"
              >
                <div
                  className="ant-row"
                  style={
                    Object {
                      "marginLeft": -4,
                      "marginRight": -4,
                      "width": "100%",
                    }
                  }
                >
                  <div
                    className="ant-col-2"
                    style={
                      Object {
                        "fontWeight": "Bold",
                        "paddingLeft": 4,
                        "paddingRight": 4,
                      }
                    }
                  >
                    <div
                      id="taskId"
                    >
                      #
                      1
                    </div>
                  </div>
                  <div
                    className="ant-col-16"
                    style={
                      Object {
                        "paddingLeft": 4,
                        "paddingRight": 4,
                      }
                    }
                  >
                    <div
                      id="taskText"
                      style={
                        Object {
                          "marginLeft": 12,
                          "marginTop": 6,
                        }
                      }
                    >
                      Go to Home
                    </div>
                  </div>
                  <div
                    className="ant-col-6"
                    style={
                      Object {
                        "paddingLeft": 4,
                        "paddingRight": 4,
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "float": "right",
                        }
                      }
                    >
                      <button
                        className="ant-btn"
                        onClick={[Function]}
                        style={
                          Object {
                            "marginRight": 7,
                          }
                        }
                        type="button"
                      >
                        <span>
                          edit
                        </span>
                      </button>
                      <button
                        className="ant-btn"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        <span>
                          delete
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
